### PR TITLE
chore(dep): upgrade to Angular 6 + rxjs 6

### PR DESCRIPTION
**Pull request by connor4312**

### CODE REVIEW

File: src/input.service.ts [hunk 1]
```diff
@@ -561,10 +560,7 @@ export class InputService {
    */
   private addKeyboardListeners() {
     this.subscriptions.push(
-      Observable.merge(
-        this.keyboardSrc,
-        Observable.fromEvent<KeyboardEvent>(window, 'keydown'),
-      ).subscribe(ev => {
+      merge(this.keyboardSrc, fromEvent<KeyboardEvent>(window, 'keydown')).subscribe(ev => {
```
Comment (irbrownie) [inline src/input.service.ts]: :'(

Comment (connor4312) [reply src/input.service.ts]: hm? Just formatting?

Comment (irbrownie) [reply src/input.service.ts]: Just that it's adding to a subs array and managing observables that way for unsubbing. no big deal.

Comment (connor4312) [reply src/input.service.ts]: ah, yea. For later refactors :)

File: src/arc.directive.ts [hunk 2]
```diff
@@ -16,7 +17,7 @@ function createDirectionCapture(direction: Direction, target: HTMLElement) {
   if (!target) {
     throw new Error(
       `Cannot set [arc-${Direction[direction]}] to an undefined element!` +
-        'Make sure the element you\'re passing is defined correctly.',
+        "Make sure the element you're passing is defined correctly.",
```
Comment (xiaoxli) [inline src/arc.directive.ts]: Just curious why use double quotes instead of single?

Comment (connor4312) [reply src/arc.directive.ts]: This is prettier's formatting -- it optimizes for least number of escape sequences in a string.

File: package.json [hunk 3]
```diff
@@ -69,7 +69,7 @@
     "ts-helpers": "^1.1.1",
     "tslint": "^5.9.1",
     "tslint-microsoft-contrib": "^5.0.3",
-    "typescript": "^2.7.2",
+    "typescript": "~2.7.2",
```
Comment (SimonSchick) [inline package.json]: Any higher version supported? Not a big deal either way.

**Comment by SimonSchick** [comment]
We probably want to look into updating the build dependencies as well, (webpack etc), might uncover some potential problems, also semver major.

File: package.json (hunk 3 above)
Comment (connor4312) [reply package.json]: nope, the compiler CLI requires 2.7 and errors if a higher version is present.

**Pull request merged**